docs/src/main/paradox/scala/http/routing-dsl/directives/marshalling-directives/handleWith.md

```diff
@@ -31,16 +31,25 @@ on marshalling entities with akka-http.
 
 ## Examples
 
-The following example uses an `updatePerson` function with a `Person` case class as an input and output.  We plug this function into our route using `handleWith`.
+The following example uses an `updatePerson` function with a `Person` @scala[case ]class as an input and output. We plug this function into our route using `handleWith`.
 
-@@snip [MarshallingDirectivesExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/directives/MarshallingDirectivesExamplesSpec.scala) { #person-case-class }
+Scala
+:   @@snip [MarshallingDirectivesExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/directives/MarshallingDirectivesExamplesSpec.scala) { #person-case-class }
 
-@@snip [MarshallingDirectivesExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/directives/MarshallingDirectivesExamplesSpec.scala) { #example-handleWith-with-json }
+Java
+:   @@snip [MarshallingDirectivesExamplesTest.java]($test$/java/docs/http/javadsl/server/directives/MarshallingDirectivesExamplesTest.java) { #person }
 
-The PersonJsonSupport object handles both marshalling and unmarshalling of the Person case class.
 
 Scala
-:  @@snip [MarshallingDirectivesExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/directives/MarshallingDirectivesExamplesSpec.scala) { #person-json-support }
+:   @@snip [MarshallingDirectivesExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/directives/MarshallingDirectivesExamplesSpec.scala) { #example-handleWith-with-json }
 
 Java
-:  @@snip [MarshallingDirectivesExamplesTest.java]($test$/java/docs/http/javadsl/server/directives/MarshallingDirectivesExamplesTest.java) { #person }
+:   @@snip [MarshallingDirectivesExamplesTest.java]($test$/java/docs/http/javadsl/server/directives/MarshallingDirectivesExamplesTest.java) { #example-handleWith-with-json }
+
+
+The @scala[PersonJsonSupport object handles]@java[previous example uses also @ref[Json Support via Jackson](../../../common/json-support.md#json-jackson-support) to handle] both marshalling and unmarshalling of the Person case class.
+
+@@@ div { .group-scala }
+@@snip [MarshallingDirectivesExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/directives/MarshallingDirectivesExamplesSpec.scala) { #person-json-support }
```

Comment:
In the current version for Java, this was listing again the `#person` snippet. So I removed it